Description: Don't ignore signals from external processes
Bug: https://bugs.launchpad.net/bugs/789389

Index: lightdm-0.3.7/src/child-process.c
===================================================================
--- lightdm-0.3.7.orig/src/child-process.c	2011-06-10 16:10:37.942724060 +1000
+++ lightdm-0.3.7/src/child-process.c	2011-06-10 16:10:34.172724062 +1000
@@ -401,12 +401,10 @@
         read (signal_pipe[0], &pid, sizeof (pid_t)) < 0)
         return TRUE;
 
-    if (pid == 0)
+    process = g_hash_table_lookup (processes, GINT_TO_POINTER (pid));
+    if (!process)
         process = child_process_get_parent ();
-    else
-        process = g_hash_table_lookup (processes, GINT_TO_POINTER (pid));
-    if (process)
-        g_signal_emit (process, signals[GOT_SIGNAL], 0, signo);
+    g_signal_emit (process, signals[GOT_SIGNAL], 0, signo);
 
     return TRUE;
 }
