=== modified file 'greeters/gtk/lightdm-example-gtk-greeter.c'
Index: lightdm-0.4.1/greeters/gtk/lightdm-example-gtk-greeter.c
===================================================================
--- lightdm-0.4.1.orig/greeters/gtk/lightdm-example-gtk-greeter.c	2011-06-30 10:59:24.000000000 +0100
+++ lightdm-0.4.1/greeters/gtk/lightdm-example-gtk-greeter.c	2011-06-30 12:29:32.815664492 +0100
@@ -474,6 +474,8 @@
     message_label = GTK_WIDGET (gtk_builder_get_object (builder, "message_label"));
     session_combo = GTK_WIDGET (gtk_builder_get_object (builder, "session_combobox"));
   
+    gtk_window_set_has_resize_grip (GTK_WINDOW (window), FALSE);
+
     gtk_label_set_text (GTK_LABEL (gtk_builder_get_object (builder, "hostname_label")), ldm_greeter_get_hostname (greeter));
 
     background_image = ldm_greeter_get_string_property (greeter, "background-image");
Index: lightdm-0.4.1/po/POTFILES.in
===================================================================
--- lightdm-0.4.1.orig/po/POTFILES.in	2011-05-24 06:35:25.000000000 +0100
+++ lightdm-0.4.1/po/POTFILES.in	2011-06-30 12:29:32.815664492 +0100
@@ -23,3 +23,4 @@
 src/xdmcp-server.c
 src/xdmcp-session.c
 src/xserver.c
+.pc/01_resize_grip.patch/greeters/gtk/lightdm-example-gtk-greeter.c
