Index: lightdm-0.9.0/tests/Makefile.am
===================================================================
--- lightdm-0.9.0.orig/tests/Makefile.am	2011-07-22 16:03:37.000000000 +1000
+++ lightdm-0.9.0/tests/Makefile.am	2011-07-22 17:04:50.109290576 +1000
@@ -49,6 +49,8 @@
 #	test-xserver-fail-start
 #	test-greeter-xserver-crash
 
+TESTS =
+
 EXTRA_DIST = \
 	$(TESTS) \
 	data/xgreeters/test-gobject-greeter.desktop \
