Description: Stop crashing due to accessing unrefed ChildProcess objects
Bug: https://bugs.launchpad.net/bugs/795050

Index: lightdm-0.3.7/src/child-process.c
===================================================================
--- lightdm-0.3.7.orig/src/child-process.c	2011-06-10 16:13:17.312723991 +1000
+++ lightdm-0.3.7/src/child-process.c	2011-06-10 16:13:16.032723992 +1000
@@ -112,8 +112,8 @@
         g_signal_emit (process, signals[TERMINATED], 0, WTERMSIG (status));
     }
 
-    g_hash_table_remove (processes, GINT_TO_POINTER (process->priv->pid));
     process->priv->pid = 0;
+    g_hash_table_remove (processes, GINT_TO_POINTER (process->priv->pid));
 }
 
 static void
@@ -323,7 +323,7 @@
 
     g_strfreev (argv);
 
-    g_hash_table_insert (processes, GINT_TO_POINTER (process->priv->pid), process);
+    g_hash_table_insert (processes, GINT_TO_POINTER (process->priv->pid), g_object_ref (process));
     g_child_watch_add (process->priv->pid, child_process_watch_cb, process);
 
     return TRUE;
@@ -453,7 +453,7 @@
                       G_TYPE_NONE, 1, G_TYPE_INT);
 
     /* Catch signals and feed them to the main loop via a pipe */
-    processes = g_hash_table_new_full (g_direct_hash, g_direct_equal, NULL, NULL);
+    processes = g_hash_table_new_full (g_direct_hash, g_direct_equal, NULL, g_object_unref);
     if (pipe (signal_pipe) != 0)
         g_critical ("Failed to create signal pipe");
     g_io_add_watch (g_io_channel_unix_new (signal_pipe[0]), G_IO_IN, handle_signal, NULL);
