=== modified file 'src/display.c'
Index: lightdm-0.9.9~bzr/src/display.c
===================================================================
--- lightdm-0.9.9~bzr.orig/src/display.c	2011-09-28 15:55:33.602339221 +1000
+++ lightdm-0.9.9~bzr/src/display.c	2011-09-28 15:55:37.974339180 +1000
@@ -467,6 +467,12 @@
         if (g_getenv ("PATH"))
             session_set_env (session, "PATH", g_getenv ("PATH"));
     }
+    else if (is_greeter)
+    {
+        gchar *t = command;
+        command = g_strdup_printf ("/usr/lib/lightdm/lightdm-greeter-session '%s'", command);
+        g_free (t);
+    }
 
     /* Variables required for regression tests */
     if (g_getenv ("LIGHTDM_TEST_STATUS_SOCKET"))
