=== modified file 'src/display.c'
Index: lightdm/src/display.c
===================================================================
--- lightdm.orig/src/display.c	2011-09-30 14:39:20.853095661 +0200
+++ lightdm/src/display.c	2011-09-30 14:39:25.921095905 +0200
@@ -439,6 +439,12 @@
             g_free (wrapper);
         }
     }
+    else if (is_greeter)
+    {
+        gchar *t = command;
+        command = g_strdup_printf ("/usr/lib/lightdm/lightdm-greeter-session '%s'", command);
+        g_free (t);
+    }
 
     /* for a guest session, run command through the wrapper covered by MAC */
     if (display->priv->autologin_guest)
