=== modified file 'src/display.c'
Index: lightdm/src/display.c
===================================================================
--- lightdm.orig/src/display.c	2011-10-25 16:59:44.225590000 -0400
+++ lightdm/src/display.c	2011-10-25 17:01:31.045046710 -0400
@@ -469,6 +469,12 @@
             g_free (wrapper);
         }
     }
+    else if (is_greeter)
+    {
+        gchar *t = command;
+        command = g_strdup_printf ("/usr/lib/lightdm/lightdm-greeter-session '%s'", command);
+        g_free (t);
+    }
 
     /* for a guest session, run command through the wrapper covered by MAC */
     if (display->priv->autologin_guest)
