=== modified file 'src/display.c'
Index: lightdm/src/display.c
===================================================================
--- lightdm.orig/src/display.c	2011-09-07 15:17:37.463555000 +1000
+++ lightdm/src/display.c	2011-09-07 16:14:19.369289500 +1000
@@ -461,6 +461,12 @@
             g_free (wrapper);
         }
     }
+    else if (is_greeter)
+    {
+        gchar *t = command;
+        command = g_strdup_printf ("/usr/lib/lightdm/lightdm-greeter-session '%s'", command);
+        g_free (t);
+    }
 
     session = DISPLAY_GET_CLASS (display)->create_session (display);
     g_return_val_if_fail (session != NULL, NULL);
