=== modified file 'src/display.c'
Index: ubuntu/src/display.c
===================================================================
--- ubuntu.orig/src/display.c	2012-01-18 10:21:41.708528000 +0100
+++ ubuntu/src/display.c	2012-01-18 11:03:49.558334295 +0100
@@ -462,6 +462,12 @@
             g_free (wrapper);
         }
     }
+    else if (is_greeter)
+    {
+        gchar *t = command;
+        command = g_strdup_printf ("/usr/lib/lightdm/lightdm-greeter-session '%s'", command);
+        g_free (t);
+    }
 
     /* for a guest session, run command through the wrapper covered by MAC */
     if (display->priv->autologin_guest)
