diff -Naur lightdm-1.2.0.orig/data/lightdm.conf lightdm-1.2.0/data/lightdm.conf
--- lightdm-1.2.0.orig/data/lightdm.conf	2012-04-29 13:59:02.000000000 +0000
+++ lightdm-1.2.0/data/lightdm.conf	2012-04-29 14:04:38.420137697 +0000
@@ -62,7 +62,7 @@
 #xdmcp-manager=
 #xdmcp-port=177
 #xdmcp-key=
-greeter-session=lightdm-greeter
+greeter-session=lightdm-webkit-greeter
 greeter-hide-users=true
 #greeter-allow-guest=true
 #greeter-show-manual-login=false
