Description: Disclose the language-selector option.
Forwarded: no
Author: Gunnar Hjalmarsson <ubuntu@gunnar.cc>

--- lightdm.orig/greeters/gtk/lightdm-gtk-greeter.conf	2011-09-20 03:45:23 +0200
+++ lightdm/greeters/gtk/lightdm-gtk-greeter.conf	2011-10-03 22:02:31 +0200
@@ -6,6 +6,7 @@
 # xft-dpi = Resolution for Xft in dots per inch (e.g. 96)
 # xft-hintstyle = What degree of hinting to use (hintnone, hintslight, hintmedium, or hintfull)
 # xft-rgba = Type of subpixel antialiasing (none, rgb, bgr, vrgb or vbgr)
+# show-language-selector (true or false)
 #
 [greeter]
 #background=
@@ -16,3 +17,4 @@
 #xft-dpi=
 #xft-hintstyle=
 #xft-rgba=
+#show-language-selector=
